samples: mesh: nrf52: minor changes to improve Mesh network performance.

vikrant8052 opened the pull request:
This edition would improve by default mesh network performance.
This is after testing with nRFMesh app from Nordic Semiconductor.

vikrant8052 committed:
samples: mesh: nrf52: edition to improve Mesh performance

This edition would improve by default mesh network
performance. This is after testing with nRFMesh app
from Nordic Semiconductor.

Signed-off-by: Vikrant More <vikrant8051@gmail.com>

vikrant8052 committed:
samples: mesh: nrf52: coding style improvements.

Removed extra blank line. Add new
bound_states_transition_type_reassignment() function in
transition.c to improve code readability & understanding.

Signed-off-by: Vikrant More <vikrant8051@gmail.com>

vikrant8052 committed:
samples: mesh: nrf52: corrected mapping of Message handler

Corrected mapping of message handler for Generic
Move Set, Generic Move Set Unack, Light CTL Set &
Light CTL Set Unack.

Signed-off-by: Vikrant More <vikrant8051@gmail.com>

jhedberg commented on samples/boards/nrf52/mesh/onoff_level_lighting_vnd_app/src/mesh/device_composition.c:
```diff
@@ -15,7 +15,7 @@
 
 static struct bt_mesh_cfg_srv cfg_srv = {
-	.relay = BT_MESH_RELAY_DISABLED,
+	.relay = BT_MESH_RELAY_ENABLED,
```
Wouldn't it make more sense to do this on a per-node basis? Ideally you shouldn't have multiple relays very close to each other since it's just redundant and will congest the available bandwidth.

vikrant8051 replied:
> .beacon = BT_MESH_BEACON_**ENABLED,**
> 
> /* 5 transmissions with 30ms interval */
> .net_transmit = BT_MESH_TRANSMIT(5, **30**),
> 
> /* 5 transmissions with 20ms interval */
> .relay_retransmit = BT_MESH_TRANSMIT(5, **20**),

As of now #nRFMesh don't support Relay On/Off feature & that is why I enable it by default.

Based on my observation after these modification in 3 lines, I saw considerable improvement in Mesh network  (GATT Bearer) performance. All nodes are more behaved in sync manner plus all NODEs are accepting commands...means no packet loss !!

Previously, Android-nRFMesh app missed the GET status response or status message broadcast by NODE after receiving reliable message & hangs forever. But now this frequency of happening is very low. This may be Android App level issue.

I know previously, when I enabled Relay on every NODE that hangs mesh network considerably.  But now things have changed. I don't know why but recent modification are most satisfactory as of now. Plus I don't need to add following lines in .... subsys/bluetooth/host/mesh/adv.c ....adv_thread()...

>                         u8_t delay;
> 
>                         BT_MESH_ADV(buf)->busy = 0;
> 
>                        /* Add 0-50ms random delay to transmissions in
>                         * order to reduce collisions.
>                         */
>                        bt_rand(&delay, sizeof(delay));
>                        k_sleep(K_MSEC(delay % 50));
> 
>                        adv_send(buf);

jhedberg commented on samples/boards/nrf52/mesh/onoff_level_lighting_vnd_app/src/mesh/device_composition.c:
```diff
@@ -32,9 +32,11 @@ static struct bt_mesh_cfg_srv cfg_srv = {
-	.net_transmit = BT_MESH_TRANSMIT(2, 20),
-	.relay_retransmit = BT_MESH_TRANSMIT(2, 20),
+	/* 5 transmissions with 30ms interval */
+	.net_transmit = BT_MESH_TRANSMIT(5, 30),
```
What's the reasoning for this? Do you have any performance measurements to justify these changes?

vikrant8051 replied:
Kindly consider above explanation.

jhedberg commented on samples/boards/nrf52/mesh/onoff_level_lighting_vnd_app/src/mesh/device_composition.c:
```diff
@@ -32,9 +32,11 @@ static struct bt_mesh_cfg_srv cfg_srv = {
+	.net_transmit = BT_MESH_TRANSMIT(5, 30),
+
+	/* 5 transmissions with 20ms interval */
+	.relay_retransmit = BT_MESH_TRANSMIT(5, 20),
```
Why does net_transmit use 30ms whereas relay_retransmit uses 20ms?

vikrant8051 replied:
Kindly consider above explanation.

vikrant8052 committed:
samples: mesh: nrf52: optimisation in timer uses

Now states which are bound have single timer that means
means for lightness variation there is one timer & for
temperature variation there is one timer.

Signed-off-by: Vikrant More <vikrant8051@gmail.com>
